app/modules/report/section/coverage.rb

```diff
@@ -0,0 +1,364 @@
+# frozen_string_literal: true
+
+module Report
+  module Section
+    module Coverage
+      extend ::Report::Section
+
+      module_function
+
+      # ! hack this entire module runs twice - once for recordings, and again
+      #   for analysis result. the :analysis_result field in options is used to
+      #   special case cte creation within the methods, until we have a better
+      #   solution like a postgres function. this is also the reason :suffix
+      #   exists - ctes need unique names. See Report::Section::Step#suffix
+
+      # ? TimeSeries.Options also required to run Coverage but this is opaque
+      # @param time_series_options [Hash] additional options to be merged
+      def options(time_series_options)
+        options = {
+          source: nil,
+          lower_field: nil,
+          upper_field: nil,
+          analysis_result: nil,
+          project_field_as: 'coverage',
+          suffix: nil
+        }
+
+        yield(options) if block_given?
+        options.compact.merge(time_series_options)
+      end
+
+      TABLE_GAP_SIZE = Arel::Table.new('gap_size')
+      TABLE_SORT_WITH_LAG = Arel::Table.new('sort_with_lag')
+      TABLE_GROUPED = Arel::Table.new('grouped')
+      TABLE_COVERAGE_INTERVALS = Arel::Table.new('coverage_intervals')
+      TABLE_COVERAGE_EVENTS = Arel::Table.new('coverage_events')
+      TABLE_COVERAGE_EVENTS_SORTED_SUM = Arel::Table.new('coverage_events_sorted_sum')
+      TABLE_COVERED_INTERVALS = Arel::Table.new('covered_intervals')
+      TABLE_COVERED_DURATIONS = Arel::Table.new('covered_durations')
+      TABLE_FINAL_COVERAGE = Arel::Table.new('final_coverage')
+
+      TABLES = {
+        gap_size: TABLE_GAP_SIZE,
+        sort_with_lag: TABLE_SORT_WITH_LAG,
+        grouped: TABLE_GROUPED,
+        coverage_intervals: TABLE_COVERAGE_INTERVALS,
+        coverage_events: TABLE_COVERAGE_EVENTS,
+        coverage_events_sorted_sum: TABLE_COVERAGE_EVENTS_SORTED_SUM,
+        covered_intervals: TABLE_COVERED_INTERVALS,
+        covered_durations: TABLE_COVERED_DURATIONS,
+        final_coverage: TABLE_FINAL_COVERAGE
+      }.freeze
```

Comment:
i've seen this structure a few times, seems a little redundant? what is it used for?

if you need a list of tables you could register them in your `step` method